Sort GraalVM Public Demos and Improve Top-Level README (#131)
graalvm/graalvm-demos: native-image-configure-examples/README.md

  image gets executed, it can run the handler right away since it was already
  determined at build time which hander should be used.

## Preparation

> 1. [Download GraalVM](https://www.graalvm.org/downloads/), unzip the archive, export the GraalVM home directory as the `$JAVA_HOME` and add `$JAVA_HOME/bin` to the `PATH` environment variable.

Reviewer: ```suggestion
1. [Download GraalVM](https://www.graalvm.org/downloads/), unzip the archive, set the GraalVM home directory as the `$JAVA_HOME` environment variable and add `$JAVA_HOME/bin` to the `PATH` environment variable.
```

    On Linux:
    ```bash
    export JAVA_HOME=/home/${current_user}/path/to/graalvm
    export PATH=$JAVA_HOME/bin:$PATH
    ```
    On macOS:
    ```bash
    export JAVA_HOME=/Users/${current_user}/path/to/graalvm/Contents/Home
    export PATH=$JAVA_HOME/bin:$PATH
    ```
    On Windows:
    ```bash
    setx /M JAVA_HOME "C:\Progra~1\Java\<graalvm>"
    setx /M PATH "C:\Progra~1\Java\<graalvm>\bin;%PATH%"
    ```

2. Install [Native Image](https://www.graalvm.org/dev/reference-manual/native-image/#install-native-image) by running:
    ```bash
    gu install native-image
    ```

    The `native-image` tool depends on some local native libraries. Ensure you meet the [platform-specific requirements](https://www.graalvm.org/dev/reference-manual/native-image/#prerequisites).

> 3. Download or clone the demos repository::

Reviewer: ```suggestion
3. Download or clone the demos repository:
```

    ```
    git clone https://github.com/graalvm/graalvm-demos
    ```

## Build and run examples

1. Change to one of the demo subdirectories, for example, _configure-at-runtime-example_:
    ```
    cd graalvm-demos/native-image-configure-examples/configure-at-runtime-example
    ```
2. Build the project:
    ```bash
    mvn package
    ```

> 3. Once the Maven build succeeds, a native executable called "example" will be generated in the _configure-at-runtime-example/target/_ folder. Execute it:

Reviewer: ```suggestion
3. Once the Maven build succeeds, a native executable called "example" will be generated in the _configure-at-runtime-example/target/_ directory. Run it:
```

    ```
    ./target/example
    Tue Mar 23 22:17:33 EET 2021
    ```
4. Repeat the same steps for the other sub-demo:
    ```
    cd ..
    cd configure-at-buildtime-example
    mvn package
    ./target/example
    Hello, world!
    ```
5. Finally, you may shutdown the `native-image` server:
    ```
    $JAVA_HOME/bin/native-image --server-shutdown
    ```

> Loading application configuration during the image build can speed up application startup. See [Class Initialization at Image Build Time](https://www.graalvm.org/dev/reference-manual/native-image/optimizations-and-performance/ClassInitialization/) to learn more. 

Reviewer: ```suggestion
Loading application configuration at executable build time can speed up application startup. See [Build-Time Initialization](https://www.graalvm.org/dev/reference-manual/native-image/optimizations-and-performance/ClassInitialization/) to learn more. 
```

> To read more about classes initialization topic, check this publication [Initialize Once, Start Fast: Application Initializationat Build Time](http://www.christianwimmer.at/Publications/Wimmer19a/Wimmer19a.pdf).

Reviewer: ```suggestion
To read more about the topic of class initialization, see [Initialize Once, Start Fast: Application Initializationat Build Time](http://www.christianwimmer.at/Publications/Wimmer19a/Wimmer19a.pdf).
```
